csc2/macc_so.c

```diff
@@ -3143,27 +3120,26 @@ static int dyns_get_field_option_comn(char *tag, int fidx, int option,
             case CLIENT_CSTR: {
                 if ((*value_type == CLIENT_CSTR || *value_type == CLIENT_PSTR ||
                      *value_type == CLIENT_VUTF8) &&
-                    vbsz >
-                        strlen(tables[tidx].sym[fidx].fopts[i].value.strval)) {
-                    bzero(valuebuf,
-                          strlen(tables[tidx].sym[fidx].fopts[i].value.strval) +
-                              1);
-                    memcpy(
-                        valuebuf, tables[tidx].sym[fidx].fopts[i].value.strval,
-                        strlen(tables[tidx].sym[fidx].fopts[i].value.strval));
-                    *value_sz =
-                        strlen(tables[tidx].sym[fidx].fopts[i].value.strval);
+                    vbsz > strlen(tables[tidx].sym[fidx].fopts[i].value.strval)) {
+                    bzero(valuebuf, strlen(tables[tidx].sym[fidx].fopts[i].value.strval) + 1);
+                    memcpy(valuebuf, tables[tidx].sym[fidx].fopts[i].value.strval,
+                           strlen(tables[tidx].sym[fidx].fopts[i].value.strval));
+                    *value_sz = strlen(tables[tidx].sym[fidx].fopts[i].value.strval);
                     return 0;
-                } else if (*value_type == CLIENT_DATETIME ||
-                           *value_type == CLIENT_DATETIMEUS) {
-                    memcpy(
-                        valuebuf, tables[tidx].sym[fidx].fopts[i].value.strval,
-                        strlen(tables[tidx].sym[fidx].fopts[i].value.strval));
-                    *value_sz =
-                        strlen(tables[tidx].sym[fidx].fopts[i].value.strval);
+                } else if (*value_type == CLIENT_DATETIME || *value_type == CLIENT_DATETIMEUS) {
+                    memcpy(valuebuf, tables[tidx].sym[fidx].fopts[i].value.strval,
+                           strlen(tables[tidx].sym[fidx].fopts[i].value.strval));
+                    *value_sz = strlen(tables[tidx].sym[fidx].fopts[i].value.strval);
                     return 0;
-                } else if (*value_type == CLIENT_BYTEARRAY &&
-                           vbsz >= tables[tidx].sym[fidx].szof) {
+                } else if (*value_type == CLIENT_BYTEARRAY && vbsz >= tables[tidx].sym[fidx].szof && 
+                           tables[tidx].sym[fidx].szof == 16 && 
+                           (strcmp(tables[tidx].sym[fidx].fopts[i].value.strval, "GUID") == 0 ||
+                           strcmp(tables[tidx].sym[fidx].fopts[i].value.strval, "GUID_FUNC") == 0)) {
+                    memcpy(valuebuf, tables[tidx].sym[fidx].fopts[i].value.strval,
+                           strlen(tables[tidx].sym[fidx].fopts[i].value.strval));
+                    *value_sz = strlen(tables[tidx].sym[fidx].fopts[i].value.strval);
```

Comment:
length can be cached in a local var.

Reply:
what happens when I do `guid byte[17] dbstore="GUID"`?

Reply:
perhaps a test for that?

Reply:
Yes, this can all be cached--but did not want to pollute in this PR. 
Updated with failure message--can actually test for that.

Reply:
I meant, you could hold `strlen(tables[tidx].sym[fidx].fopts[i].value.strval))` into a var instead of calling it twice.

Reply:
oh, good point. thanks.

Reply:
Done.